src/test/java/ReaderWriterTest.java

```diff
@@ -0,0 +1,66 @@
+package test;
+
+import IO.DbReader;
+import IO.DbWriter;
+import data.Database;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+/* Validate that a .txt file can be properly be stored written and read from database.  */
+public final class ReaderWriterTest {
+  private static String inputFile;
+  private static String outputFile;
+  private static InputStream input;
+  private static OutputStream output;
+
+  private Database database = new Database("test");
+  private DbWriter writer = new DbWriter(database, "test", "writingIn");
+  private DbReader reader = new DbReader(database, "test", "writingIn");
+
+  @Before
+  public void setUp() {}
+
+  @Test
+  public void validateDatabaseName() {
+    System.out.println("validateDatabaseName()");
+    /* Validate that place where data is stored is named test. */
+    System.out.println("test = " + database.getDatabase());
+    Assert.assertEquals(database.getDirectoryName(), "test");
+  }
+
+  @Test
+  public void validateWriterRunId() {
+    System.out.println("validateWriterRunId()");
+    /* Validate that writer's runId is test. */
+    Assert.assertEquals(writer.getRunId(), "test");
+  }
+
+  @Test
+  public void validateReaderWriterDatabaseConnect() throws IOException {
+    System.out.println("validateReaderWriterDatabaseConnect()");
```

Comment:
any suggestions for testing? this one had errors and i was unsure of how to solve them. I was hoping to store the string "This is a test string". into my database and read and write from it. I wanted to test if the stream input and output return was equal, but the code did not compile.

Reply:
What was the issue with compilation, that seems like a reasonable suggestion?

Reply:
update: it compiles now! there was an issue with one of the DbWriter's constructor